### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by p-galligan**
Migrates from Travis to GHA

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      name: ${{ github.ref_name }}
+
+    env:
+      APPLICATION_NAME: argo
+      CONTAINER: argo-web
+      APPLICATION_PORT: 8014
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```
Comment (helrond) [inline .github/workflows/deploy.yml:23]: I think it might be necessary to add submodules here too.

**Review by helrond — CHANGES_REQUESTED** [review]
Just one small thing - I think we need to pull in submodules for deploy as well because that code needs to be available to the app.

Branch rules also need to be updated.

**Comment by helrond** [comment]
@p-galligan Branch protection rules need to be updated.

**Comment by p-galligan** [comment]
> @p-galligan Branch protection rules need to be updated.

Thanks! Got them.

**Review by HaSistrunk — APPROVED** [review]

**Review by helrond — APPROVED** [review]

**Pull request merged**